sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -243,8 +243,17 @@ async def do_get():
 
             monitor = settings.get_monitor()
             inputs = await rpc.serialize_properties({"urn": urn}, {})
+
+            accept_resources = not (
+                os.getenv("PULUMI_DISABLE_RESOURCE_REFERENCES", "").upper()
+                in {"TRUE", "1"}
+            )
```

Comment:
Its debatable whether we even need to continue respecting `PULUMI_DISABLE_RESOURCE_REFERENCES` (especially in these new cases), which was used when we were originally rolling out resource refs as a way to be able to turn off the feature if something went wrong during the rollout. We still check the envvar elsewhere, so I'm continuing to use it in these new places. Separately, we should consider removing it throughout as it's not really needed anymore (resource refs have been working fine E2E -- there hasn't been a need to disable them).

Reply:
Lets go with this as is for now. Sounds like a good thing to do a cleanup pass on though.